com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java

```diff
@@ -487,9 +497,28 @@ private static File createAppBundleBaseZip(Project project, File outDir, File ap
                 BundleHelper.throwIfCanceled(canceled);
             }
 
-            // copy assets
+            // copy extension and bundle resoources
             Map<String, IResource> bundleResources = ExtenderUtil.collectBundleResources(project, getArchitectures(project));
-            ExtenderUtil.writeResourcesToDirectory(bundleResources, assetsDir);
+            final String assets = "assets" + File.separator;
+            for (String filename : bundleResources.keySet()) {
+                IResource resource = bundleResources.get(filename);
+                // remove initial file separator if it exists
+                if (filename.startsWith(File.separator)) {
+                    filename = filename.substring(1);
+                }
+                // files starting with "assets/" should be copied to the assets/ dir
+                // other files should be copied to the to the root/ dir
+                File file = null;
+                if (filename.startsWith(assets)) {
+                    file = new File(assetsDir, filename.substring(assets.length()));
+                }
+                else {
+                    file = new File(rootDir, filename);
+                }
+                log("Copying resource to " + file);
+                ExtenderUtil.writeResourceToFile(resource, file);
+            }
```

Comment:
The actual fix. Files starting with assets/ should be copied to the assets/ dir and files starting with something else should be copied to the root/ dir.